### PR TITLE
Adds view, languages and menu

### PR DESCRIPTION
**Pull request by FinalAngel**
- This has been recreated from `feature/pagetree` as it was corrupt.
- Please ignore `cms/static/cms/js/libs/jstree/`
- Be aware that there are still lots of comments and consoles (todos)

Review the following sections closely:
- https://github.com/FinalAngel/django-cms/pull/83/files#diff-8081c2cc590f5172aab625068c9e67deR1
- https://github.com/FinalAngel/django-cms/pull/83/files#diff-c0f49b7fb03324e7ffa911fddf188faaR1
- https://github.com/FinalAngel/django-cms/pull/83/files#diff-87a4c90dd47124c284c73a4a30f3a864R1
- https://github.com/FinalAngel/django-cms/pull/83/files#diff-acb64d300cdba9435f9841649c554e7bL1

image

image

### CODE REVIEW

File: cms/static/cms/js/modules/cms.pagetree.js [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+    // shorthand for jQuery(document).ready();
+    $(function () {
+
+        // shifts options element inside the node
+        $.jstree.plugins.options = function (options, parent) {
+            var buttons = '.cms-tree-data';
+
+            // what happens when redrawing
+            this.redraw_node = function (obj, deep, callback, force_draw) {
+                // default draw ,ethod
```
Comment (dhorelik) [inline cms/static/cms/js/modules/cms.pagetree.js]: `method`, a typo

File: cms/static/cms/sass/components/_pagetree.scss [hunk 2]
```diff
@@ -0,0 +1,198 @@
+//##############################################################################
+// PAGETREE
+
+$pagetree-icon-size: 18px;
+$pagetree-lang-size: 14px;
+$pagetree-tooltip-min-width: 130px;
+
+.cms-tree {
+    .cms-icon {
+        font-family: django-cms-iconfont;
```
Comment (Brankub) [inline cms/static/cms/sass/components/_pagetree.scss]: where is $? or I missing something

File: cms/static/cms/sass/components/_pagetree.scss [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+}
+
+//###########################################################
+// PREVIEW
+// shows preview eye icon
+.cms-tree-preview {
+    position: relative;
+    top: 1px;
+}
+.cms-tree-preview a {
```
Comment (dhorelik) [inline cms/static/cms/sass/components/_pagetree.scss]: nesting? 
```
.cms-tree-preview {
    a {...}
}
```

File: cms/static/cms/sass/components/_pagetree.scss [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+
+    span {
+        display: inline-block;
+        @include hide-content();
+    }
+}
+// active state for current selected page
+.cms-tree-preview-active a:before {
+    color: $black;
+}
```
Comment (Brankub) [inline cms/static/cms/sass/components/_pagetree.scss]: maybe better to use `&-active a:before` before?

File: cms/static/cms/sass/components/_pagetree.scss [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+        animation: pulsate 2.5s ease-out infinite;
+        border-color: $color-primary;
+        background-color: $color-primary;
+    }
+}
+
+// animation for publishing
+@keyframes pulsate {
+    0% {
+        opacity: 0.5;
```
Comment (dhorelik) [inline cms/static/cms/sass/components/_pagetree.scss]: `opacity: 0;`?

File: cms/templates/admin/cms/page/tree/base.html [hunk 6]
```diff
@@ -1,126 +1,174 @@
...
 
-{% include "admin/cms/page/tree/base_header.html" %}
-
-</div>
-</div>
-
-<div id="dialogs"></div>
+{% block result_list %}
+    <div class="clear"></div>
+    <p>Page Tree</p>
```
Comment (Brankub) [inline cms/templates/admin/cms/page/tree/base.html]: trans?

File: cms/static/cms/js/modules/cms.pagetree.js [hunk 7]
```diff
@@ -0,0 +1,242 @@
...
+                var triggers = $('.js-cms-tree-tooltip-trigger');
+                var containers = $('.js-cms-tree-tooltip-container');
+
+                // attach event to the trigger
+                triggers.on('click.pagetree', function (e) {
+                    e.preventDefault();
+                    e.stopImmediatePropagation();
+
+                    containers.removeClass('cms-tree-tooltip-container-open');
+                    containers.eq(triggers.index(this))
```
Comment (dhorelik) [inline cms/static/cms/js/modules/cms.pagetree.js]: `.removeClass().eq()` are chainable

File: cms/static/cms/sass/components/_pagetree.scss (hunk 2 above)
Comment (FinalAngel) [inline cms/static/cms/sass/components/_pagetree.scss]: that file is included from within sass/cms.pagetree.scss where we load ``@import "settings/all";``

File: cms/templates/admin/cms/page/tree/base.html (hunk 6 above)
Comment (FinalAngel) [inline cms/templates/admin/cms/page/tree/base.html]: this will be removed later on

File: cms/templates/admin/cms/page/tree/base.html [hunk 8]
```diff
@@ -1,126 +1,174 @@
...
+        "csrf": "{{ csrf_token }}",
+        "static": "{{ STATIC_URL }}",
+        "permission": "{{ CMS_PERMISSION|bool }}",
+        "debug": "{{ DEBUG|bool }}",
+        "filtered": "{{ cl.is_filtered|bool }}",
+        "lang": {
+            "success": "{% filter escapejs %}{% trans "Successfully moved" %}{% endfilter %}",
+            "changes": "{% filter escapejs %}{% trans "Changes within the tree might require a refresh." %}{% endfilter %}",
+            "error": "{% filter escapejs %}{% trans "An error occured. Please reload the page" %}{% endfilter %}",
+            "publish": "{% filter escapejs %}{% trans "Are you sure you want to § this page?" %}{% endfilter %}",
```
Comment (dhorelik) [inline cms/templates/admin/cms/page/tree/base.html]: why the `§` sign here?

File: cms/static/cms/sass/components/_pagetree.scss [hunk 9]
```diff
@@ -0,0 +1,199 @@
...
+.jstree-grid-cell {
+    overflow: visible !important;
+}
+.jstree-grid-header {
+    font-size: $font-size-small;
+    text-align: center;
+}
+.jstree-grid-column-0 .jstree-grid-header {
+    text-align: left;
+}
```
Comment (dhorelik) [inline cms/static/cms/sass/components/_pagetree.scss]:
```
.jstree-grid-header {
    ...
    .jstree-grid-column-0 & {
        text-align: left;
    }
}
```

File: cms/static/cms/sass/components/_pagetree.scss (hunk 2 above)
Comment (Brankub) [inline cms/static/cms/sass/components/_pagetree.scss]: in this case it should be in quotes, no? as here https://github.com/divio/django-cms/blob/7828aa1f3c82c1bdb6168dc006d6c759cce3d704/cms/static/cms/sass/components/_iconography.scss#L6

File: cms/templates/admin/cms/page/tree/base.html (hunk 6 above)
Comment (Brankub) [inline cms/templates/admin/cms/page/tree/base.html]: :+1:

File: cms/templates/admin/cms/page/tree/base.html (hunk 8 above)
Comment (FinalAngel) [inline cms/templates/admin/cms/page/tree/base.html]: This is used for replacement: §

**Comment by Brankub** [comment]
:+1:

File: cms/templates/admin/cms/page/tree/base.html (hunk 8 above)
Comment (dhorelik) [inline cms/templates/admin/cms/page/tree/base.html]: aha, I see, thanks

**Comment by dhorelik** [comment]
:+1:

**Pull request merged**